### PR TITLE
[FLINK-28200] Add table store Hive reader documentation

### PR DESCRIPTION
**Pull request by tsreaper**
This PR adds documentation for table store Hive reader.

### CODE REVIEW

File: docs/content/docs/engines/hive.md [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+Copy table store Hive connector bundle jar to a path accessible by Hive, then use `add jar /path/to/hive-connector.jar` to enable table store support in Hive.
+
+## Using Table Store Hive Catalog
+
+By using table store Hive catalog, you can create, drop and insert into table store tables from Flink. These operations directly affect the corresponding Hive metastore. Tables created in this way can also be accessed directly from Hive.
+
+Execute the following Flink SQL script in Flink SQL client to define a table store Hive catalog and create a table store table.
+
+```sql
+-- Define table store Hive catalog
```
Comment (JingsongLi) [inline docs/content/docs/engines/hive.md:63]: Although it is explained above, I feel that it is still possible to confuse Flink CLI and Hive CLI.
I think it is better to re-emphasis on Flink SQL here. Like:
```
-- Flink SQL CLI
-- Define table store Hive catalog
```

File: docs/content/docs/engines/hive.md [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+
+```bash
+mvn clean install -DskipTests
+```
+
+You can find the bundle jar in `"./flink-table-store-hive/flink-table-store-hive-connector/target/flink-table-store-hive-connector-{{< version >}}.jar"`.
+
+{{< /unstable >}}
+
+Copy table store Hive connector bundle jar to a path accessible by Hive, then use `add jar /path/to/hive-connector.jar` to enable table store support in Hive.
```
Comment (JingsongLi) [inline docs/content/docs/engines/hive.md:54]: `hive-connector.jar` => `flink-table-store-hive-connector-{{< version >}}.jar`

File: docs/content/docs/engines/hive.md [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+KIND, either express or implied.  See the License for the
+specific language governing permissions and limitations
+under the License.
+-->
+
+# Hive
+
+Table Store currently supports the following features related with Hive:
+* Create, drop and insert into table store tables through table store Hive catalog. Tables created in this way can also be read directly from Hive.
```
Comment (JingsongLi) [inline docs/content/docs/engines/hive.md:30]: In Flink SQL, create, drop ....

File: docs/content/docs/engines/hive.md [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+* Register existing table store tables as external tables in Hive.
```
Comment (JingsongLi) [inline docs/content/docs/engines/hive.md:31]: In Hive SQL, ....

**Comment by tsreaper** [comment]
> I think it's better to separate HiveCatalog (Used by Flink) and Hive Reader (Used by Hive) into two pages.

Let's do this in a separate PR. I'd like to have a dedicated catalog page to introduce each catalog.

**Review by JingsongLi — APPROVED** [review]
Looks good to me!

**Pull request merged**